### PR TITLE
[api] enable DIAG_GET query function into commissioner

### PR DESCRIPTION
**Pull request by ZhangLe2016**
This PR adds DIAG_GET query command into the commissioner module, allowing users to query diagnostic TLV (Ipv6Address) from a peer Thread device by setting the device's mesh local address than TLV mask bits flag.

Test:
config set pskc 445f2b5ca6f2a93a55ce570a70efeecb
[done]
start 192.168.9.2 49154
[done]
diag query ipaddr fdce:bb1d:37f6:6978:0:ff:fe00:cc00
Peer Address: fd74:95e3:6ec:feee:0:ff:fe00:ac00
Content: {
    "Ipv6 Addresses": [
        "fd74:95e3:06ec:feee:0000:00ff:fe00:fc36",
        "fd74:95e3:06ec:feee:0000:00ff:fe00:fc11",
        "fd74:95e3:06ec:feee:0000:00ff:fe00:fc10",
        "fd74:95e3:06ec:feee:0000:00ff:fe00:fc38",
        "fd11:0022:0000:0000:99e7:cece:d50d:ae6b",
        "fd74:95e3:06ec:feee:0000:00ff:fe00:fc00",
        "fd74:95e3:06ec:feee:0000:00ff:fe00:ac00",
        "fd74:95e3:06ec:feee:6142:033d:b087:2845",
        "fe80:0000:0000:0000:9cbb:e247:df46:9e70"
    ]
}
[done]

### CODE REVIEW

File: include/commissioner/commissioner.hpp [hunk 1]
```diff
@@ -259,6 +260,19 @@ class CommissionerHandler
...
     }
 
+    /**
+     * This function notifies the receiving the queried Diagnostic TLVs by DIAG_GET.ans command.
+     *
+     * @param[in] aDestAddress  The destination address of the DIAG_GET.ans command.
+     * @param[in] aDiagAnsMsg   Parsed DiagTlvs data.
+     *
+     */
+    virtual void OnDiagGetAnswerMessage(const std::string &aDestAddress, const NetDiagData &aDiagAnsMsg)
```
[the remark this answers is not in the document]
Comment (ZhangLe2016) [reply include/commissioner/commissioner.hpp]: done

File: include/commissioner/network_diag_data.hpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+#define IPV6_ADDRESS_BYTES 16
+
+namespace ot {
+
+namespace commissioner {
+
+/**
+ * @brief IPv6 Addresses List
+ */
+struct Ipv6AddressList
```
[the remark this answers is not in the document]
Comment (ZhangLe2016) [reply include/commissioner/network_diag_data.hpp:55]: this data structure will be reused into the `ChildIpv6Address`, which includes the child's `rloc16` and `Ipv6AddressList` two fields,  we could reuse this data structure decode function into the `ChildIpv6Address` .

File: include/commissioner/network_diag_data.hpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+};
+
+/**
+ * @brief network diagnostic data in TMF
+ *
+ * Each data field of Diagnostic TLVs is optional. The field is
+ * meaningful only when associative PresentFlags is included in
+ * `mPresentFlags`.
+ */
+struct NetDiagData
```
[the remark this answers is not in the document]
Comment (ZhangLe2016) [reply include/commissioner/network_diag_data.hpp:67]: the previous PR is too large with all possible fields, can we split them into some small PR for code review?

File: include/commissioner/network_diag_data.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (wgtdkp) [reply include/commissioner/network_diag_data.hpp:55]: Why can't `ChildIpv6Address` use `std::vector<std::string>` directly?

File: include/commissioner/network_diag_data.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (wgtdkp) [reply include/commissioner/network_diag_data.hpp:67]: Sure.

But it's better to review and submit the NetDiagData definition and decoding before providing an API to request for NetDiagData. Let me know when you get the NetDiagData class PR ready.

I am stopping reviewing this CL now.

File: include/commissioner/network_diag_data.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ZhangLe2016) [reply include/commissioner/network_diag_data.hpp:55]: I think it is more reasonable for `Ipv6AddressList` as a sub structure of `ChildIpv6Address` and `NetDiagData`.

File: include/commissioner/network_diag_data.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (ZhangLe2016) [reply include/commissioner/network_diag_data.hpp:67]: okay, let me provide a new PR for it.

[the remark this answers is not in the document]
Comment (ZhangLe2016) [reply include/commissioner/network_diag_data.hpp:67]: new PR #303 is ready for your comments, Thanks.

**Comment by ZhangLe2016** [comment]
closed this PR and created a new PR #304

**Pull request closed without merging**